🧪IMPROVE: Replaces Arjan's videos with new testing content

MrCordeiro opened the pull request:
Part of the feedback we have received is that Arjan's videos were too complex.

Makes sense: Arjan focuses a lot on system design and OOP, even when writing tests, and beginners are not fully ready for that yet.

# Changes

- Replaced Arjan's videos with 3 resources: an overall intro to Pytest, a video focused on mock.patch and one extra resource for fixtures
- Moved all the content from the Word document to a markdown file
- Expanded the testing dataframe document with comments and a better comparison helper function

RoberVega reviewed (changes requested):
- Section 1:
  - On the main README.md it would be nice to have a link to the second section on pyproject too as we have for the first one

- Testing:
  - Testing Dataframes:
      - Give a little note on what from pathlib does on the first example
      - On the second example, give a reason why you switch from pytest to unittest: from unittest.mock import patch, Mock (more on this later on)
      - On Pyspark Dataframes explain what Logging does and why you have some functions with an underscore in the beginning of the function. 
      - 10 minutes reading seems too little. I would give around 30 to go through the functions.
      - I prefer the pytest documentation. It is what one would check as a first option most of the time so it is good to get familiar with the contents and I just like it more, think it is clearer.
      - “What should you test?” section could be moved before the Pandas Dataframe tests or even at the beginning of the document to introduce the concept of private functions.
      - Maybe even move the extra content before as well give the monkeypatch explanation before  and therefore motivate the usage of unittest within the Dataframe testing file.

MrCordeiro commented:
> Section 1:
>
> On the main README.md it would be nice to have a link to the second section on pyproject too as we have for the first one

I don't quite understand what you mean...

> Give a little note on what from pathlib does on the first example

By this point (testing), I think the student should have already seen that. They appear in the earlier assignments...

> On the second example, give a reason why you switch from pytest to unittest: from unittest.mock import patch, Mock (more on this later on)

We didn't. That is still a pytest test. This is aligned with the videos, that use pytest together with Mock's from unittest.

> On Pyspark Dataframes explain what Logging does and why you have some functions with an underscore at the beginning of the function.

Perhaps it would be better to include that in modules of their own. I removed the logs so they are not distracting.

> 10 minutes reading seems too little. I would give around 30 to go through the functions.

Changed! 

> I prefer the pytest documentation. It is what one would check as a first option most of the time so it is good to get familiar with the contents and I just like it more, think it is clearer.

Done!

> “What should you test?” section could be moved before the Pandas Dataframe tests or even at the beginning of the document to introduce the concept of private functions.

Done!

> Maybe even move the extra content before as well give the monkeypatch explanation before and therefore motivate the usage of unittest within the Dataframe testing file.

Moving extra content to the top gives off the impression is not really "extra". I don't understand, however, what do you mean with `motivate the usage of unittest`... as none of the videos cover monkeypatch, is very likely a student with no prior knowledge will only know how to patch using unittest.mock objects at this point.

RoberVega reviewed (approved):
Everything seems good now!
